sdk/storage/examples/connection_string.rs

```diff
@@ -61,16 +61,16 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
         .execute()
         .await?;
 
-    println!("List blob returned {} blobs.", iv.incomplete_vector.len());
-    for cont in iv.incomplete_vector.iter() {
-        println!("\t{}\t{} bytes", cont.name, cont.content_length);
+    println!("List blob returned {} blobs.", iv.blobs.blobs.len());
```

Comment:
Why repeated? I mean, clearly they are different types, but I can't tell from this limited view why the word itself is repeated. If we ever do a arch review with our board (and I think we should before first GA, at least a sampling) this may raise some questions.

Reply:
Yes that's ugly (I was referring about this in the the PR description). 

I copied the XML structure defined here: [https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs#response-body](https://docs.microsoft.com/en-us/rest/api/storageservices/list-blobs#response-body) is like this:

```xml
<EnumerationResults ServiceEndpoint="http://myaccount.blob.core.windows.net/"  ContainerName="mycontainer">  
  <Blobs>  
     <Blob>...</Blob>
     <Blob>...</Blob>
     <Blob>...</Blob>
     ...
     <BlobPrefix>  
        <Name>blob-prefix</Name>  
     </BlobPrefix> 
     <BlobPrefix>  
        <Name>blob-prefix</Name>  
     </BlobPrefix> 
```

So that's why the code ends up like this: `response.blobs.blobs` with the last `blobs` being the blob array. 

What do you think would be better?

Reply:
In the existing Azure SDKs, I believe the first `blobs` here would actually be a `BlobClient` that you get from a `BlobContainerClient`. The second `blobs` I understand is the collection of blobs. This is one area where language SDKs diverge a lot and "blobs" may be idiomatic here. For example, in .NET from a `BlobClient` you would call `GetBlobs()` which returns an `IEnumerable<T>`- there's never really a name. Some other languages are similar, but often use `listBlobs()` instead because it's idiomatic. I've seen in Rust (and same in Go, I believe, and we do have a few SDKs there) basically what you did here: a method that is basically the collection, i.e. `blobs`. So, I suppose, my suggestion is like I questioned below about the `as_` methods: if we get "sub-clients" (Account -> Container -> Blob, or Account -> Queue, etc.) similar to other Azure SDKs, this might not be an issue.